homeassistant/components/switch/rest.py

```diff
@@ -47,21 +54,27 @@ def setup_platform(hass, config, add_devices_callback, discovery_info=None):
         _LOGGER.error("No route to resource/endpoint: %s", resource)
         return False
 
```

Comment:
We have changed the way templates work. Templates will be passed in as Template objects (because of `cv.template` validation) and you only have to set hass on it.

```python
value_template.hass = hass
```

Then, instead of calling `template.render(hass, template, …)` you can do `self.value_template.render(…)`